rpcs3/Emu/System.cpp

```diff
@@ -2977,6 +2977,21 @@ void Emulator::Kill(bool allow_autoexit, bool savestate)
 
 game_boot_result Emulator::Restart(bool graceful)
 {
+	if (m_state == system_state::stopping)
+	{
+		// Emulation stop is in progress
+		return game_boot_result::still_running;
+	}
+
+	Emu.after_kill_callback = [this]
```

Comment:
I pushed a change.
This is to make Restart potentially async, the above check is for preventing a race with competing kill.

Reply:
Since Restart() is async now, should it use `GracefulShutdown(async_op=true)`?

Reply:
GracefulShutdown does it a different, it waits for the process to kill itself.